client/client.go

```diff
@@ -116,26 +81,10 @@ func (c *Client) DialAny(ski gokeyless.SKI) (*gokeyless.Conn, error) {
 		return nil, fmt.Errorf("no servers registered for SKI %02x", ski)
 	}
 
-	for _, server := range servers {
-		c.m.RLock()
-		conn, ok := c.conns[server]
-		c.m.RUnlock()
-		if ok {
-			if conn.Use() {
-				return conn, nil
-			}
-			c.m.Lock()
-			if c.conns[server] == conn {
-				delete(c.conns, server)
-			}
-			c.m.Unlock()
-		}
-	}
-
 	// choose from possible servers at random until a connection can be established.
```

Comment:
This is a bit of a punt. Really there should only be one `Remote` registered. We could enforce that by having `registerSKI` append to the current remote (creating a `group` if necessary).